ReactAndroid/src/main/java/com/facebook/react/views/imagehelper/ResourceDrawableIdHelper.java

```diff
@@ -66,12 +71,40 @@ public int getResourceDrawableId(Context context, @Nullable String name) {
     }
   }
 
+  private String sanitizeResourceDrawableId(@NonNull String name) {
+    return name.toLowerCase().replace("-", "_");
+  }
+
   public @Nullable Drawable getResourceDrawable(Context context, @Nullable String name) {
+    if (name == null || name.isEmpty()) {
+      return null;
+    }
+
+    name = sanitizeResourceDrawableId(name);
+
+    // Checks to see if we have an ota version of the file, otherwise default to normal behavior.
+    File otaFile = ResourceDrawableUtils.getDrawableFileFromCache(context, name);
+    if (otaFile != null) {
+      return Drawable.createFromPath(otaFile.getAbsolutePath());
+    }
```

Comment:
IMO move this chunk as a static method in `ResourceDrawableUtils` or better yet have something like (pseudocode).

```
static class ImageOTAUtils {

   fun getOTADrawableOrNull()
   fun getOTAUriOrNull()

   class ResourceDrawableUtils {
   }
   enum PhoneDensity {
   } 
}
```

Now there will only be two lines changes in this stock RN file and only a single "new" file in our branch that we can easily delete/update in the future.

Reply:
don't want to clutter this file for sure, but having three separate files that will never have conflicts is about as bad as having one giant file so i'll split things up for readability